### PR TITLE
CatManga QoL improvements

### PR DESCRIPTION
**Pull request by ivaniskandar**
One endpoint for all operations except for fetching pages. Should be faster when refreshing library.

### CODE REVIEW

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt [hunk 1]
```diff
@@ -15,11 +15,12 @@ import org.json.JSONArray
 import org.json.JSONObject
 import org.jsoup.nodes.Document
 import rx.Observable
-import uy.kohesive.injekt.Injekt
-import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
 
 class CatManga : HttpSource() {
 
+    private val application: Application by injectLazy()
```
Comment (CarlosEsco) [inline src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:22]: Can't you inject the source pref directly similar to 
https://github.com/tachiyomiorg/tachiyomi-extensions/blob/master/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt#L43

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt [hunk 2]
```diff
@@ -202,6 +226,14 @@ class CatManga : HttpSource() {
...
+        throw UnsupportedOperationException("Not used.")
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        throw UnsupportedOperationException("Not used.")
+    }
+
     override fun searchMangaParse(response: Response): MangasPage {
         throw UnsupportedOperationException("Not used.")
     }
```
Comment (CarlosEsco) [inline src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: I think in general we are trying to switch extensions from gson and the internal json to kotlinx.  So if you could do that in this PR

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt (hunk 1 above)
Comment (ivaniskandar) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:22]: I don't think so, the shared prefs is different for each manga. https://github.com/tachiyomiorg/tachiyomi-extensions/pull/7451/files#diff-f22fc81b84173dcff99a1c03c2f56b42717f74c380fd06982a9d5616be88c2f8R84

Comment (CarlosEsco) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:22]: hmm why is the shared preference different per a manga and not just a different entry?

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt (hunk 2 above)
Comment (ivaniskandar) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: It doesn't use any gson stuff here. Both [JSONArray](https://developer.android.com/reference/org/json/JSONArray) and [JSONObject](https://developer.android.com/reference/org/json/JSONObject) are available in Android's SDK.

I've tried switching to kotlinx.serialization without data classes and the code became way less readable. And as for now with the current JSON format in their site I don't know how to do it with data class without any workarounds.

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt (hunk 1 above)
Comment (ivaniskandar) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:22]: Should've request the change when I first implemented this 😅 But okay I'll make it populated inside single prefs

Comment (CarlosEsco) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:22]: sorry i didn't notice its fine how it is then.

File: src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt (hunk 2 above)
Comment (ObserverOfTime) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: > I've tried switching to kotlinx.serialization without data classes and the code became way less readable.

I 100% agree with this statement.

Comment (h-hyuuga) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: kotlinx migration for catmanga is already done @CarlosEsco 

https://github.com/tachiyomiorg/tachiyomi-extensions/pull/7407

Comment (CarlosEsco) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: @h-hyuuga that will probably need a slight refactor after I merge this

Comment (ivaniskandar) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: I'll try to pick @h-hyuuga changes to my branch as discussed

Comment (h-hyuuga) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: Don't, someone else already is

Comment (ivaniskandar) [reply src/en/catmanga/src/eu/kanade/tachiyomi/extension/en/catmanga/CatManga.kt:239]: Okay then I'll mark this PR ready.

**Pull request merged**